src/core/main.c

```diff
@@ -1721,17 +1721,55 @@ static void update_numa_policy(bool skip_setup) {
                 log_warning_errno(r, "Failed to set NUMA memory policy: %m");
 }
 
+static void filter_args(const char* dst[], unsigned *pos, char **src, int argc) {
+        /* Copy some filtered arguments into the dst array from src. */
```

Comment:
`assert` on some of the inputs?

Reply:
I added assert on the first two args. `src` and `argc` can be NULL or negative, and we should be fine.

Reply:
maybe also `assert(argc == 0 || src)`? We do that often iirc